dev/src/pool.ts

```diff
@@ -42,6 +42,12 @@ export class ClientPool<T> {
    */
   private terminated = false;
 
+  /**
+   * Deferred promise that is resolved when there are no active operations on
+   * the client pool.
+   */
+  private operationsCompletedDeferred = new Deferred<void>();
```

Comment:
Ideally, we would also block new tasks from submission. We could then use a single boolean ("terminatePending") and a deferred Promise ("terminateDeferred"), which might make this code a bit cleaner. "Operations Completed" is a state that happens all the time, but it is only relevant during shutdown.

Reply:
The `terminated` boolean already blocks new tasks from submission after terminate() is called. Renamed to `terminateDeferred`